packages/core/src/combinator/combine.ts

```diff
@@ -50,21 +55,27 @@ class Combine {
   }
 }
 
-class CombineSink extends Pipe {
-  constructor (disposables, sinks, sink, f) {
+class CombineSink<A, B> extends Pipe<B | IndexedValue<A>> {
+  private readonly disposables: Disposable[]
+  private readonly f: (...args: A[]) => B
+  private awaiting: number
+  private hasValue: boolean[]
+  private activeCount: number
+  private readonly values: A[]
+
+  constructor (disposables: Disposable[], length: number, sink: Sink<B>, f: (...args: A[]) => B) {
     super(sink)
     this.disposables = disposables
-    this.sinks = sinks
     this.f = f
 
-    const l = sinks.length
+    const l = length
```

Comment:
This seems redundant now.

Reply:
Why? The length is used in constructing `values` and `hasValue` arrays

Reply:
Because you are just assigning `l` the value of `length`. Why not just use `length` in the code below instead of `l`? While this code is running, two variables must now be kept in memory.

Reply:
The reason `sources.length` _was_ assigned to `l` was an optimization instead of accessing the `length` property several times on the Array object. Besides, if the `sinks` array were mutated (passed by reference to the constructor), the length property would change.

Reply:
oops, you are right